drivers/filesystems/fastfat/misc.c

```diff
@@ -416,23 +416,43 @@ VfatLockUserBuffer(
 BOOLEAN
 VfatCheckForDismount(
     IN PDEVICE_EXTENSION DeviceExt,
-    IN BOOLEAN Create)
+    IN BOOLEAN Force)
 {
     KIRQL OldIrql;
+    ULONG UnCleanCount = 3; // FIXME: or 2 depending on whether we are in IRP_MJ_CREATE; however I think we still leak like 1 or 2 extra refs there...
     PVPB Vpb;
     BOOLEAN Delete;
 
-    DPRINT1("VfatCheckForDismount(%p, %u)\n", DeviceExt, Create);
+    DPRINT1("VfatCheckForDismount(%p, %u)\n", DeviceExt, Force);
+
+    /* If the VCB is OK (not under uninitialization) and we don't force dismount, do nothing */
+    if (BooleanFlagOn(DeviceExt->Flags, VCB_GOOD) && !Force)
+    {
+        return FALSE;
+    }
+
+#if 0 // See the other 3rd-party IFSes, or fastfat_new that uses instead a "ResidualOpenCount".
```

Comment:
What's the point of this?

Reply:
That's something we may need to consider later (see the comment in the commit message), and I would like to have your advice on this.

Reply:
I believe it won't make it to our implementation. It's future is to die and to be replaced by MS FastFAT, not to have ground breaking changes. So, just drop it.